File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemFilterDemoVModel.java
package io.ganguo.demo.viewmodel.item;

import android.databinding.ObservableField;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ItemFilterDemoBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.view.AdapterInterface;

/**
 * Created by leo on 2016/12/21.
 */
public class ItemFilterDemoVModel extends BaseViewModel<AdapterInterface<ItemFilterDemoBinding>> {
    private ObservableField<String> title = new ObservableField<>();

    public ItemFilterDemoVModel(String title) {
        this.title.set(title);
    }

    public ObservableField<String> getTitle() {
        return title;
    }

    public void setTitle(ObservableField<String> title) {
        this.title = title;
    }


    @Override
    public String toString() {
        return title.get();
    }

    @Override
    public void onViewAttached(View view) {

    }

    @Override
    public int getItemLayoutId() {
        return R.layout.item_filter_demo;
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemFilterDemoVModel.kt
package io.ganguo.demo.viewmodel.item

import androidx.databinding.ObservableField
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemFilterDemoBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.AdapterInterface

/**
 * Created by leo on 2016/12/21.
 */
class ItemFilterDemoVModel(title: String) : BaseViewModel<io.ganguo.viewmodel.core.viewinterface.AdapterInterface<ItemFilterDemoBinding>>() {
    override val layoutId: Int by lazy { R.layout.item_filter_demo }
    var title = ObservableField<String>()

    init {
        this.title.set(title)
    }


    override fun onViewAttached(view: View) {

    }


}
